### ISSUE TITLE
Failed to fetch stream on web

### ISSUE THREAD

**Issue by Zia-Bashir** [Zekfad/fetch_client#7]
Hi, I am having trouble fetching the stream on Flutter Web. I tried a lot, but it didn't work

```
log("FETCH CLIENT ++++++++++++++++++");
        final client = FetchClient(
          mode: RequestMode.cors,
          streamRequests: true,
        );

        final uri =
            Uri.parse("https://my uri");
        final stream = (() async* {
          yield Uint8List.fromList(
            '''
      {
        "name": "My cool data",
        "data": {
          "data_part_1": "part_1",
      '''
                .codeUnits,
          );
          await Future<void>.delayed(const Duration(seconds: 1));
          yield Uint8List.fromList(
            '''
          "data_part_2": "part_2"
        }
      }
      '''
                .codeUnits,
          );
        })();

        final request = StreamedRequest('POST', uri)
          ..headers.addAll({
            "Authorization": "Bearer $idToken",
            'content-type': 'application/json',
          })
          ..sink.add(utf8.encode(json.encode(body)));
         // body is the map of data that I've to pass with post request
        
        stream.listen(
          request.sink.add,
          onDone: request.sink.close,
          onError: request.sink.addError,
        );

        final response = await client.send(request);

        log(await utf8.decodeStream(response.stream));

        return;
```
**Log Error:  error Failed to execute fetch: TypeError: Failed to fetch**

**Comment by Zekfad** [comment]
It is supported only in Chrome 105+ and requires http/2 or http/3 web server to accept the request.
I'd suggest you to use web socket of you need outgoing streaming with more browsers support.

**Comment by Zia-Bashir** [comment]
Hi Zekfad,
We solved the issue. Now we get the streaming but the issue is where we can pass the body parameter. In backend we're getting null in body and body contains uid,and list of messages. You can see in code. It'll be helpful if you'll tell me the correct way. Thanks

**Comment by Zekfad** [comment]
The code from above seems ok.
I suspect that issue in the server. You may be using reverse proxy, like nginx which could create buffer.
Code from example folder works, that implies request is ok.
I suggest you to try http2 tools to test your server, if it can accept streamed request.

**Comment by k00na** [comment]
Hey, @Zekfad ! 

I would like to report that I have also stumbled on the same limitation when using Flutter web and trying to receive streams. 

What's confusing is also that I have found this article on medium which describes the limitations of Http for web and streaming but then proposes fetch_client library as solution to this limitation. And now I'm confused. Does this library solve the streaming issue for Flutter web or not? 

https://medium.com/@islam_akhrarov/title-the-limitations-of-the-dart-http-library-and-a-solution-for-web-streaming-574818b360d9

Thank you for your time 🙏
